modules/apps/list-type/list-type-test/src/testIntegration/java/com/liferay/list/type/service/test/ListTypeEntryLocalServiceTest.java

```diff
@@ -115,39 +151,28 @@ public void testAddListTypeEntry() throws Exception {
 			_listTypeEntry.getUuid(),
 			_listTypeEntry.getExternalReferenceCode());
 
-		// Duplicate key
+		// No ListTypeDefinition exists with the primary key
 
 		try {
-			_testAddListTypeEntry(
-				_listTypeDefinition.getListTypeDefinitionId(), "able");
+			_testAddListTypeEntry(0, "able");
+
+			Assert.fail();
 		}
-		catch (DuplicateListTypeEntryException
-					duplicateListTypeEntryException) {
+		catch (NoSuchListTypeDefinitionException
+					noSuchListTypeDefinitionException) {
 
 			Assert.assertEquals(
-				"Duplicate key able",
-				duplicateListTypeEntryException.getMessage());
+				"No ListTypeDefinition exists with the primary key 0",
+				noSuchListTypeDefinitionException.getMessage());
 		}
 
-		String externalReferenceCode =
-			_listTypeEntry.getExternalReferenceCode();
-
-		try {
-			_listTypeEntryLocalService.addListTypeEntry(
-				externalReferenceCode, TestPropsValues.getUserId(),
-				_listTypeDefinition.getListTypeDefinitionId(),
-				RandomTestUtil.randomString(),
-				Collections.singletonMap(
-					LocaleUtil.US, RandomTestUtil.randomString()));
-		}
-		catch (DuplicateListTypeEntryExternalReferenceCodeException
-					duplicateListTypeEntryExternalReferenceCodeException) {
+		// System picklist
 
-			Assert.assertEquals(
-				"Duplicate external reference code " + externalReferenceCode,
-				duplicateListTypeEntryExternalReferenceCodeException.
-					getMessage());
-		}
+		AssertUtils.assertFailure(
+			ListTypeDefinitionSystemException.class, false,
+			"Only allowed bundles can add system list type entries",
+			() -> _testAddListTypeEntry(
+				_systemListTypeDefinition.getListTypeDefinitionId(), "able"));
```

Comment:
You should put another key here, since `able` already exists. Otherwise, if the test fails (if we can create this entry), we would face a duplicate key exception. Am I missing anything?